### PR TITLE
Add duzun/base64util type definitions

### PR DESCRIPTION
**Pull request by ghost**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/base64util/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      "strictNullChecks": true,
+      "strictFunctionTypes": true,
+      "baseUrl": "../",
+      "typeRoots": [
+          "../"
+      ],
+      "types": [],
+      "noEmit": true,
+      "forceConsistentCasingInFileNames": true,
+      "esModuleInterop": true
```
Comment (gabritto) [inline types/base64util/tsconfig.json:18]: Can you remove `esModuleInterop` from this config? We try to adhere to a standard set of compiler options in DT whenever possible, and this option is not in the standard.

File: types/base64util/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+export function mbDecode(data: string): string;
+export function mbEncode(data: string): string;
+export function mbUrlDecode(data: string): string;
+export function mbUrlEncode(data: string): string;
+export function polyfill(data: string): string;
+export function urlDecode(data: string): string;
+export function urlEncode(data: string): string;
+export function utf82mb(data: string): string;
+
+export function bindProto(proto: BASE64.Prototype): void;
```
Comment (gabritto) [inline types/base64util/index.d.ts:33]: Does this function show up in the original library? I couldn't find it when looking at the library's source (https://unpkg.com/browse/base64util@2.2.0/base64.js).

**Review by gabritto — CHANGES_REQUESTED** [review]
Looks good overall, I left comments about small changes to make before I approve this. Thanks.

File: types/base64util/index.d.ts (hunk 2 above)
Comment (mdaemon-technologies) [reply types/base64util/index.d.ts:33]: https://github.com/duzun/base64util/blob/master/src/base64.js

It shows up here. Not sure why it wouldn't show up in unpkg.com.

Maybe even though it's being exported, it's only used internally?

File: types/base64util/tsconfig.json (hunk 1 above)
Comment (mdaemon-technologies) [reply types/base64util/tsconfig.json:18]: For sure. I'll update it asap.

File: types/base64util/index.d.ts (hunk 2 above)
Comment (gabritto) [reply types/base64util/index.d.ts:33]: Thanks! Indeed, this function shows up on the package but for some reason not in the corresponding file in `unpkg.com`. Weird.

File: types/base64util/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// Type definitions for base64util 2.2
+// Project: https://github.com/duzun/base64util
+// Definitions by: Jonathan Ehman <https://github.com/jpehman>
```
Comment (gabritto) [inline types/base64util/index.d.ts:3]: Is this your personal account? Can you confirm this is the github account you'll be using for e.g. create/review future PRs to this package?

Comment (mdaemon-technologies) [reply types/base64util/index.d.ts:3]: Yes @jpehman is my personal account. I will use it to contribute and update this package in the future.

**Review by gabritto — APPROVED** [review]

**Comment by jpehman** [comment]
Ready to merge.

Thanks for your time 🙂

**Comment by mdaemon-technologies** [comment]
Just in case

Ready to merge.

**Comment by gabritto** [comment]
It seems the bot is having issues with merging PRs... @typescript-bot ready to merge

**Pull request merged**